src/views/Patient/MedReqDropDown/MedReqDropDown.tsx

```diff
@@ -135,8 +135,21 @@ function MedReqDropDown(props: any) {
 
 
     useEffect(() => {
-        if (patient && patient.id && client.user.id && selectedMedicationCardBundle) {
-            const hook = new OrderSign(patient.id, client.user.id, { resourceType: 'Bundle', type: 'batch', entry: [selectedMedicationCardBundle] });
+        // get the userId from the client if possible, otherwise pull it from the appContext
+        let userId = client.user.id;
+        if (!userId) {
+            const appContextString = client.state?.tokenResponse?.appContext;
+            console.log('appContext: ' + appContextString);
+            const appContext: { [key: string]: string } = {};
+            appContextString.split('&').map((e: string)=>{
+                const temp: string[] = e.split('=');
+                appContext[temp[0]] = temp[1];
+            });
+            userId = appContext?.user;
```

Comment:
the userId should be a part of the state if it is going to be possibly absent from the props (through the client).  This process of analyzing the appContext for the userId should take place in a useEffect like the one that gets the patient and sets the patient state, so that we can reuse the userId in other places it might be needed.

Reply:
I have updated this, please see the changes that I have made.